src/app/components/inputnumber/inputnumber.ts

```diff
@@ -98,6 +98,7 @@ export const INPUTNUMBER_VALUE_ACCESSOR: any = {
                 [attr.data-pc-section]="'input'"
                 pAutoFocus
                 [autofocus]="autofocus"
+                [class.p-variant-filled]="variant == 'filled'"
```

Comment:
In here, could you please also check the inputStyle signal in config? As an example, you can check autocomplete component: https://github.com/primefaces/primeng/blob/8446c5b3f896e40d99f7b9a2a626284c2c107580/src/app/components/autocomplete/autocomplete.ts#L816